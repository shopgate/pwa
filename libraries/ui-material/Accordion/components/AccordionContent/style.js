import { css } from 'glamor';

export const content = css({
  overflow: 'hidden',
  willChange: 'height',
});

export const contentInner = css({
  padding: '0 16px 16px',
  overflow: 'hidden',
}).toString();
