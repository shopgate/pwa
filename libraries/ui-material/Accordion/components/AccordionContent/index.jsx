import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import * as styles from './style';

/**
 * The accordion content component.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function AccordionContent({
  children, open, id, className,
}) {
  const ref = useRef(null);
  const height = (ref.current === null) ? 'auto' : ref.current.clientHeight;
  const style = {
    height: !open ? 0 : height,
  };

  return (
    <div className={styles.content} style={style} id={id} aria-hidden={!open}>
      <div ref={ref} className={classNames(styles.contentInner, className)}>
        {children}
      </div>
    </div>
  );
}

AccordionContent.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  open: PropTypes.bool,
};

AccordionContent.defaultProps = {
  className: '',
  open: false,
};

export default AccordionContent;
