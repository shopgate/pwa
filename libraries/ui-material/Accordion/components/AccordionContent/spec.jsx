import React from 'react';
import { mount } from 'enzyme';
import AccordionContent from './index';

jest.mock('react-spring', () => ({
  ...jest.requireActual('react-spring'),
  useSpring: jest.fn().mockReturnValue({ hook: 'useSpring return value' }),
}));

describe('<AccordionContent />', () => {
  it('should render as closed', () => {
    const wrapper = mount((
      <AccordionContent id="some-id">
        <div id="test">Some Child</div>
      </AccordionContent>
    ));

    expect(wrapper.find('#test').text()).toEqual('Some Child');
    expect(wrapper.find('div').get(0).props['aria-hidden']).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render as open', () => {
    const wrapper = mount((
      <AccordionContent open id="some-id">
        <div id="test">Some Child</div>
      </AccordionContent>
    ));

    expect(wrapper.find('#test').text()).toEqual('Some Child');
    expect(wrapper.find('div').get(0).props['aria-hidden']).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });
});
