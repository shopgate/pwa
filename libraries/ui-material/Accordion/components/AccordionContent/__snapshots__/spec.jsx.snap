// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionContent /> should render as closed 1`] = `
<div
  aria-hidden={true}
  className={
    Object {
      "data-css-apm51k": "",
    }
  }
  id="some-id"
  style={
    Object {
      "height": 0,
    }
  }
>
  <div
    className="css-yh2lvw"
  >
    <div
      id="test"
    >
      Some Child
    </div>
  </div>
</div>
`;

exports[`<AccordionContent /> should render as open 1`] = `
<div
  aria-hidden={false}
  className={
    Object {
      "data-css-apm51k": "",
    }
  }
  id="some-id"
  style={
    Object {
      "height": "auto",
    }
  }
>
  <div
    className="css-yh2lvw"
  >
    <div
      id="test"
    >
      Some Child
    </div>
  </div>
</div>
`;
